### Project 1: Data Cleaning for NLP1000 ###
#### Group 19: Mansukhani, Asanro Jay E. ####

The bible texts were scraped from bible.com and biblegateway.com via a custom web scraper I created for each website. 
Since both websites didn't contain a complete collection for the required 8 + 2N languages which in my case is 10 languages, the cleaning process will differ depending on which website it was scraped from.

In [32]:
import importlib
import utils

importlib.reload(utils)

from utils import *


### Counting Raw Corpus Size of Languages ###

In [34]:
raw_folder = "Bible/Raw/"
sizes = count_corpus_size(raw_folder)

for lang, count in sizes.items():
    print(f"{lang}: {count} words")

ABCEB - Cebuano: 103101 words
ABTAG2001 - Tagalog: 102137 words
HLGN - Ilonggo: 100321 words
KJ21 - English: 100741 words
MBBBIK92 - Bikol: 88120 words
MBBPAN83 - Pangasinense: 87751 words
MBBSAM - Waray: 97387 words
NVI - Spanish: 91130 words
PMPV - Kapampangan: 91813 words
RIPV - Ilokano: 77601 words


- ABCEB - Cebuano: 103101 words
- ABTAG2001 - Tagalog: 102137 words
- HLGN - Ilonggo: 100321 words
- KJ21 - English: 100741 words
- MBBBIK92 - Bikol: 88120 words
- MBBPAN83 - Pangasinense: 87751 words
- MBBSAM - Waray: 97387 words
- NVI - Spanish: 91130 words
- PMPV - Kapampangan: 91813 words
- RIPV - Ilokano: 77601 words

### Cleaning Tagalog Language Books ###
Tagalog book was scraped from biblegateway.com

Exodus

In [35]:
tagalog_raw = "Bible/Raw/ABTAG2001 - Tagalog/Exodo"
tagalog_cleaned = "Bible/Cleaned/Tagalog/Exodus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(tagalog_raw,tagalog_cleaned,expressions) 


Cleaned: Exodo_1.txt
Cleaned: Exodo_10.txt
Cleaned: Exodo_11.txt
Cleaned: Exodo_12.txt
Cleaned: Exodo_13.txt
Cleaned: Exodo_14.txt
Cleaned: Exodo_15.txt
Cleaned: Exodo_16.txt
Cleaned: Exodo_17.txt
Cleaned: Exodo_18.txt
Cleaned: Exodo_19.txt
Cleaned: Exodo_2.txt
Cleaned: Exodo_20.txt
Cleaned: Exodo_21.txt
Cleaned: Exodo_22.txt
Cleaned: Exodo_23.txt
Cleaned: Exodo_24.txt
Cleaned: Exodo_25.txt
Cleaned: Exodo_26.txt
Cleaned: Exodo_27.txt
Cleaned: Exodo_28.txt
Cleaned: Exodo_29.txt
Cleaned: Exodo_3.txt
Cleaned: Exodo_30.txt
Cleaned: Exodo_31.txt
Cleaned: Exodo_32.txt
Cleaned: Exodo_33.txt
Cleaned: Exodo_34.txt
Cleaned: Exodo_35.txt
Cleaned: Exodo_36.txt
Cleaned: Exodo_37.txt
Cleaned: Exodo_38.txt
Cleaned: Exodo_39.txt
Cleaned: Exodo_4.txt
Cleaned: Exodo_40.txt
Cleaned: Exodo_5.txt
Cleaned: Exodo_6.txt
Cleaned: Exodo_7.txt
Cleaned: Exodo_8.txt
Cleaned: Exodo_9.txt


Genesis

In [36]:
tagalog_raw = "Bible/Raw/ABTAG2001 - Tagalog/Genesis"
tagalog_cleaned = "Bible/Cleaned/Tagalog/Genesis"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(tagalog_raw,tagalog_cleaned,expressions) 


Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [38]:
tagalog_raw = "Bible/Raw/ABTAG2001 - Tagalog/Levitico"
tagalog_cleaned = "Bible/Cleaned/Tagalog/Leviticus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(tagalog_raw,tagalog_cleaned,expressions) 


Cleaned: Levitico_1.txt
Cleaned: Levitico_10.txt
Cleaned: Levitico_11.txt
Cleaned: Levitico_12.txt
Cleaned: Levitico_13.txt
Cleaned: Levitico_14.txt
Cleaned: Levitico_15.txt
Cleaned: Levitico_16.txt
Cleaned: Levitico_17.txt
Cleaned: Levitico_18.txt
Cleaned: Levitico_19.txt
Cleaned: Levitico_2.txt
Cleaned: Levitico_20.txt
Cleaned: Levitico_21.txt
Cleaned: Levitico_22.txt
Cleaned: Levitico_23.txt
Cleaned: Levitico_24.txt
Cleaned: Levitico_25.txt
Cleaned: Levitico_26.txt
Cleaned: Levitico_27.txt
Cleaned: Levitico_3.txt
Cleaned: Levitico_4.txt
Cleaned: Levitico_5.txt
Cleaned: Levitico_6.txt
Cleaned: Levitico_7.txt
Cleaned: Levitico_8.txt
Cleaned: Levitico_9.txt


### Cleaning Cebuano Language Books ###
Cebuano book was scraped from bible.com

Exodus

In [ ]:
cebuano_raw = "Bible/Raw/ABCEB - Cebuano/Exodus"
cebuano_cleaned = "Bible/Cleaned/Cebuano/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(cebuano_raw,cebuano_cleaned,expressions)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt


Genesis

In [3]:
cebuano_raw = "Bible/Raw/ABCEB - Cebuano/Genesis"
cebuano_cleaned = "Bible/Cleaned/Cebuano/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(cebuano_raw,cebuano_cleaned,expressions)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [24]:
cebuano_raw = "Bible/Raw/ABCEB - Cebuano/Leviticus"
cebuano_cleaned = "Bible/Cleaned/Cebuano/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(cebuano_raw,cebuano_cleaned,expressions)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt


### Cleaning Hiligaynon Language Books ###
Hiligaynon books were scraped from biblegateway.com

Exodus

In [18]:
hlgn_raw = "Bible/Raw/HLGN - Ilonggo/Exodus"
hlgn_cleaned = "Bible/Cleaned/Hiligaynon/Exodus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(hlgn_raw,hlgn_cleaned,expressions) 

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt


In [19]:
clean_verses_in_folder(hlgn_cleaned) #Fix verse numbering bug

✔ Cleaned Verses Exodus_1.txt
✔ Cleaned Verses Exodus_10.txt
✔ Cleaned Verses Exodus_11.txt
✔ Cleaned Verses Exodus_12.txt
✔ Cleaned Verses Exodus_13.txt
✔ Cleaned Verses Exodus_14.txt
✔ Cleaned Verses Exodus_15.txt
✔ Cleaned Verses Exodus_16.txt
✔ Cleaned Verses Exodus_17.txt
✔ Cleaned Verses Exodus_18.txt
✔ Cleaned Verses Exodus_19.txt
✔ Cleaned Verses Exodus_2.txt
✔ Cleaned Verses Exodus_20.txt
✔ Cleaned Verses Exodus_21.txt
✔ Cleaned Verses Exodus_22.txt
✔ Cleaned Verses Exodus_23.txt
✔ Cleaned Verses Exodus_24.txt
✔ Cleaned Verses Exodus_25.txt
✔ Cleaned Verses Exodus_26.txt
✔ Cleaned Verses Exodus_27.txt
✔ Cleaned Verses Exodus_28.txt
✔ Cleaned Verses Exodus_29.txt
✔ Cleaned Verses Exodus_3.txt
✔ Cleaned Verses Exodus_30.txt
✔ Cleaned Verses Exodus_31.txt
✔ Cleaned Verses Exodus_32.txt
✔ Cleaned Verses Exodus_33.txt
✔ Cleaned Verses Exodus_34.txt
✔ Cleaned Verses Exodus_35.txt
✔ Cleaned Verses Exodus_36.txt
✔ Cleaned Verses Exodus_37.txt
✔ Cleaned Verses Exodus_38.txt
✔ Cleaned V

Genesis

In [37]:
hlgn_raw = "Bible/Raw/HLGN - Ilonggo/Genesis"
hlgn_cleaned = "Bible/Cleaned/Hiligaynon/Genesis"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(hlgn_raw,hlgn_cleaned,expressions) 

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

In [38]:
clean_verses_in_folder(hlgn_cleaned)  #Fix verse numbering bug

✔ Cleaned Genesis_1.txt
✔ Cleaned Genesis_10.txt
✔ Cleaned Genesis_11.txt
✔ Cleaned Genesis_12.txt
✔ Cleaned Genesis_13.txt
✔ Cleaned Genesis_14.txt
✔ Cleaned Genesis_15.txt
✔ Cleaned Genesis_16.txt
✔ Cleaned Genesis_17.txt
✔ Cleaned Genesis_18.txt
✔ Cleaned Genesis_19.txt
✔ Cleaned Genesis_2.txt
✔ Cleaned Genesis_20.txt
✔ Cleaned Genesis_21.txt
✔ Cleaned Genesis_22.txt
✔ Cleaned Genesis_23.txt
✔ Cleaned Genesis_24.txt
✔ Cleaned Genesis_25.txt
✔ Cleaned Genesis_26.txt
✔ Cleaned Genesis_27.txt
✔ Cleaned Genesis_28.txt
✔ Cleaned Genesis_29.txt
✔ Cleaned Genesis_3.txt
✔ Cleaned Genesis_30.txt
✔ Cleaned Genesis_31.txt
✔ Cleaned Genesis_32.txt
✔ Cleaned Genesis_33.txt
✔ Cleaned Genesis_34.txt
✔ Cleaned Genesis_35.txt
✔ Cleaned Genesis_36.txt
✔ Cleaned Genesis_37.txt
✔ Cleaned Genesis_38.txt
✔ Cleaned Genesis_39.txt
✔ Cleaned Genesis_4.txt
✔ Cleaned Genesis_40.txt
✔ Cleaned Genesis_41.txt
✔ Cleaned Genesis_42.txt
✔ Cleaned Genesis_43.txt
✔ Cleaned Genesis_44.txt
✔ Cleaned Genesis_45.txt
✔ Cl

Leviticus

In [34]:
hlgn_raw = "Bible/Raw/HLGN - Ilonggo/Leviticus"
hlgn_cleaned = "Bible/Cleaned/Hiligaynon/Leviticus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(hlgn_raw,hlgn_cleaned,expressions) 

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt


In [35]:
clean_verses_in_folder(hlgn_cleaned)  #Fix verse numbering bug

Cleaned Verses lev 1.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Leviticus_18.txt
Cleaned Verses Leviticus_19.txt
Cleaned Verses Leviticus_2.txt
Cleaned Verses Leviticus_20.txt
Cleaned Verses Leviticus_21.txt
Cleaned Verses Leviticus_22.txt
Cleaned Verses Leviticus_23.txt
Cleaned Verses Leviticus_24.txt
Cleaned Verses Leviticus_25.txt
Cleaned Verses Leviticus_26.txt
Cleaned Verses Leviticus_27.txt
Cleaned Verses Leviticus_3.txt
Cleaned Verses Leviticus_4.txt
Cleaned Verses Leviticus_5.txt
Cleaned Verses Leviticus_6.txt
Cleaned Verses Leviticus_7.txt
Cleaned Verses Leviticus_8.txt
Cleaned Verses Leviticus_9.txt


In [36]:
#Realign verses 
align_verses("Leviticus",[4,8,11,12,14,15,16,17,18,20,22,23,24,25,26,27], hlgn_cleaned)

Aligned Leviticus_4.txt
Aligned Leviticus_8.txt
Aligned Leviticus_11.txt
Aligned Leviticus_12.txt
Aligned Leviticus_14.txt
Aligned Leviticus_15.txt
Aligned Leviticus_16.txt
Aligned Leviticus_17.txt
Aligned Leviticus_18.txt
Aligned Leviticus_20.txt
Aligned Leviticus_22.txt
Aligned Leviticus_23.txt
Aligned Leviticus_24.txt
Aligned Leviticus_25.txt
Aligned Leviticus_26.txt
Aligned Leviticus_27.txt


### Cleaning English Language Books ###
These books were scraped from biblegateway.com

Exodus

In [43]:
english_raw = "Bible/Raw/KJ21 - English/Exodus"
english_cleaned = "Bible/Cleaned/English/Exodus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(english_raw,english_cleaned,expressions) 

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt


Genesis

In [44]:
english_raw = "Bible/Raw/KJ21 - English/Genesis"
english_cleaned = "Bible/Cleaned/English/Genesis"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(english_raw,english_cleaned,expressions) 

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [45]:
english_raw = "Bible/Raw/KJ21 - English/Leviticus"
english_cleaned = "Bible/Cleaned/English/Leviticus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(english_raw,english_cleaned,expressions) 

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt


### Cleaning Bikol Language Books ###
These books were scraped from bible.com, scraping from this website was cleaner because of their layout so there are almost 0 regEx Expressions.

Exodus

In [4]:
bikol_raw = "Bible/Raw/MBBBIK92 - Bikol/Exodus"
bikol_cleaned = "Bible/Cleaned/Bikol/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(bikol_raw,bikol_cleaned,expressions)
clean_verses_in_folder(bikol_cleaned)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt
✔ Cleaned Exodus_1.txt
✔ Cleaned Exodus_10.txt
✔ Cleaned Exodus_11.txt
✔ Cleaned Exodus_1

Genesis

In [40]:
bikol_raw = "Bible/Raw/MBBBIK92 - Bikol/Genesis"
bikol_cleaned = "Bible/Cleaned/Bikol/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(bikol_raw,bikol_cleaned,expressions)
clean_verses_in_folder(bikol_cleaned)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [41]:
bikol_raw = "Bible/Raw/MBBBIK92 - Bikol/Leviticus"
bikol_cleaned = "Bible/Cleaned/Bikol/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(bikol_raw,bikol_cleaned,expressions)
clean_verses_in_folder(bikol_cleaned)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Levit

### Cleaning Pangasinense Books ###
These books were scraped from bible.com

Exodus

In [45]:

png_raw = "Bible/Raw/MBBPAN83 - Pangasinense/Exodus"
png_cleaned = "Bible/Cleaned/Pangasinense/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(png_raw,png_cleaned,expressions)
clean_verses_in_folder(png_cleaned)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt
Cleaned Verses Exodus_1.txt
Cleaned Verses Exodus_10.txt
Cleaned Verses Exodus_11.txt
Cle

Genesis 

In [46]:

png_raw = "Bible/Raw/MBBPAN83 - Pangasinense/Genesis"
png_cleaned = "Bible/Cleaned/Pangasinense/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(png_raw,png_cleaned,expressions)
clean_verses_in_folder(png_cleaned)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [47]:
png_raw = "Bible/Raw/MBBPAN83 - Pangasinense/Leviticus"
png_cleaned = "Bible/Cleaned/Pangasinense/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(png_raw,png_cleaned,expressions)
clean_verses_in_folder(png_cleaned)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Levit

### Cleaning Waray Language Books ###
These books were scraped from bible.com

Exodus

In [24]:
waray_raw = "Bible/Raw/MBBSAM - Waray/Exodus"
waray_cleaned = "Bible/Cleaned/Waray/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(waray_raw,waray_cleaned,expressions)
clean_verses_in_folder(waray_cleaned)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt
Cleaned Verses Exodus_1.txt
Cleaned Verses Exodus_10.txt
Cleaned Verses Exodus_11.txt
Cle

Genesis

In [25]:
waray_raw = "Bible/Raw/MBBSAM - Waray/Genesis"
waray_cleaned = "Bible/Cleaned/Waray/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(waray_raw,waray_cleaned,expressions)
clean_verses_in_folder(waray_cleaned)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [26]:
waray_raw = "Bible/Raw/MBBSAM - Waray/Leviticus"
waray_cleaned = "Bible/Cleaned/Waray/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(waray_raw,waray_cleaned,expressions)
clean_verses_in_folder(waray_cleaned)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Levit

### Cleaning Spanish Language Books ###
These books were scraped from biblegateway.com, these were easy to clean since every verse was pretty much 1:1

Exodus

In [15]:
spanish_raw = "Bible/Raw/NVI - Spanish/Exodus"
spanish_cleaned = "Bible/Cleaned/Spanish/Exodus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(spanish_raw,spanish_cleaned,expressions) 

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt


Genesis

In [16]:
spanish_raw = "Bible/Raw/NVI - Spanish/Genesis"
spanish_cleaned = "Bible/Cleaned/Spanish/Genesis"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(spanish_raw,spanish_cleaned,expressions) 

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [17]:
spanish_raw = "Bible/Raw/NVI - Spanish/Leviticus"
spanish_cleaned = "Bible/Cleaned/Spanish/Leviticus"

expressions = [
    (r"\(\s*[A-Za-z0-9]+\s*\)",""), #remove parenthesis
    (r"\[\s*\w\s*\]",""),           #remove footnotes
    (r"Read the Bible\s*Study Tools\s*Bible Gateway Plus\s*Explore More\s*Store", ""), #remove erroneous scrape
    (r"^[A-Za-zÁÉÍÓÚÑÜ].*(?=\r?\n\s*\d+\s)","") #remove chapter title
]

clean_text_folder(spanish_raw,spanish_cleaned,expressions) 


Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt


### Cleaning Kapampangan Language Books ###
These books were scraped from bible.com, so minimal cleaning methods will be applied

Exodus

In [18]:
pmpg_raw = "Bible/Raw/PMPV - Kapampangan/Exodus"
pmpg_cleaned = "Bible/Cleaned/Kapampangan/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(pmpg_raw,pmpg_cleaned,expressions)
clean_verses_in_folder(pmpg_cleaned)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt
Cleaned Verses Exodus_1.txt
Cleaned Verses Exodus_10.txt
Cleaned Verses Exodus_11.txt
Cle

Genesis

In [19]:
pmpg_raw = "Bible/Raw/PMPV - Kapampangan/Genesis"
pmpg_cleaned = "Bible/Cleaned/Kapampangan/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(pmpg_raw,pmpg_cleaned,expressions)
clean_verses_in_folder(pmpg_cleaned)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [20]:
pmpg_raw = "Bible/Raw/PMPV - Kapampangan/Leviticus"
pmpg_cleaned = "Bible/Cleaned/Kapampangan/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(pmpg_raw,pmpg_cleaned,expressions)
clean_verses_in_folder(pmpg_cleaned)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Levit

### Cleaning Ilokano Language Books ###
Books were scraped from bible.com

Exodus

In [21]:
ilokano_raw = "Bible/Raw/RIPV - Ilokano/Exodus"
ilokano_cleaned = "Bible/Cleaned/Ilokano/Exodus"

expressions = [
] 

clean_text_folder_no_flatten(ilokano_raw,ilokano_cleaned,expressions)
clean_verses_in_folder(ilokano_cleaned)

Cleaned: Exodus_1.txt
Cleaned: Exodus_10.txt
Cleaned: Exodus_11.txt
Cleaned: Exodus_12.txt
Cleaned: Exodus_13.txt
Cleaned: Exodus_14.txt
Cleaned: Exodus_15.txt
Cleaned: Exodus_16.txt
Cleaned: Exodus_17.txt
Cleaned: Exodus_18.txt
Cleaned: Exodus_19.txt
Cleaned: Exodus_2.txt
Cleaned: Exodus_20.txt
Cleaned: Exodus_21.txt
Cleaned: Exodus_22.txt
Cleaned: Exodus_23.txt
Cleaned: Exodus_24.txt
Cleaned: Exodus_25.txt
Cleaned: Exodus_26.txt
Cleaned: Exodus_27.txt
Cleaned: Exodus_28.txt
Cleaned: Exodus_29.txt
Cleaned: Exodus_3.txt
Cleaned: Exodus_30.txt
Cleaned: Exodus_31.txt
Cleaned: Exodus_32.txt
Cleaned: Exodus_33.txt
Cleaned: Exodus_34.txt
Cleaned: Exodus_35.txt
Cleaned: Exodus_36.txt
Cleaned: Exodus_37.txt
Cleaned: Exodus_38.txt
Cleaned: Exodus_39.txt
Cleaned: Exodus_4.txt
Cleaned: Exodus_40.txt
Cleaned: Exodus_5.txt
Cleaned: Exodus_6.txt
Cleaned: Exodus_7.txt
Cleaned: Exodus_8.txt
Cleaned: Exodus_9.txt
Cleaned Verses Exodus_1.txt
Cleaned Verses Exodus_10.txt
Cleaned Verses Exodus_11.txt
Cle

Genesis

In [22]:
ilokano_raw = "Bible/Raw/RIPV - Ilokano/Genesis"
ilokano_cleaned = "Bible/Cleaned/Ilokano/Genesis"

expressions = [
] 

clean_text_folder_no_flatten(ilokano_raw,ilokano_cleaned,expressions)
clean_verses_in_folder(ilokano_cleaned)

Cleaned: Genesis_1.txt
Cleaned: Genesis_10.txt
Cleaned: Genesis_11.txt
Cleaned: Genesis_12.txt
Cleaned: Genesis_13.txt
Cleaned: Genesis_14.txt
Cleaned: Genesis_15.txt
Cleaned: Genesis_16.txt
Cleaned: Genesis_17.txt
Cleaned: Genesis_18.txt
Cleaned: Genesis_19.txt
Cleaned: Genesis_2.txt
Cleaned: Genesis_20.txt
Cleaned: Genesis_21.txt
Cleaned: Genesis_22.txt
Cleaned: Genesis_23.txt
Cleaned: Genesis_24.txt
Cleaned: Genesis_25.txt
Cleaned: Genesis_26.txt
Cleaned: Genesis_27.txt
Cleaned: Genesis_28.txt
Cleaned: Genesis_29.txt
Cleaned: Genesis_3.txt
Cleaned: Genesis_30.txt
Cleaned: Genesis_31.txt
Cleaned: Genesis_32.txt
Cleaned: Genesis_33.txt
Cleaned: Genesis_34.txt
Cleaned: Genesis_35.txt
Cleaned: Genesis_36.txt
Cleaned: Genesis_37.txt
Cleaned: Genesis_38.txt
Cleaned: Genesis_39.txt
Cleaned: Genesis_4.txt
Cleaned: Genesis_40.txt
Cleaned: Genesis_41.txt
Cleaned: Genesis_42.txt
Cleaned: Genesis_43.txt
Cleaned: Genesis_44.txt
Cleaned: Genesis_45.txt
Cleaned: Genesis_46.txt
Cleaned: Genesis_47.

Leviticus

In [23]:
ilokano_raw = "Bible/Raw/RIPV - Ilokano/Leviticus"
ilokano_cleaned = "Bible/Cleaned/Ilokano/Leviticus"

expressions = [
] 

clean_text_folder_no_flatten(ilokano_raw,ilokano_cleaned,expressions)
clean_verses_in_folder(ilokano_cleaned)

Cleaned: Leviticus_1.txt
Cleaned: Leviticus_10.txt
Cleaned: Leviticus_11.txt
Cleaned: Leviticus_12.txt
Cleaned: Leviticus_13.txt
Cleaned: Leviticus_14.txt
Cleaned: Leviticus_15.txt
Cleaned: Leviticus_16.txt
Cleaned: Leviticus_17.txt
Cleaned: Leviticus_18.txt
Cleaned: Leviticus_19.txt
Cleaned: Leviticus_2.txt
Cleaned: Leviticus_20.txt
Cleaned: Leviticus_21.txt
Cleaned: Leviticus_22.txt
Cleaned: Leviticus_23.txt
Cleaned: Leviticus_24.txt
Cleaned: Leviticus_25.txt
Cleaned: Leviticus_26.txt
Cleaned: Leviticus_27.txt
Cleaned: Leviticus_3.txt
Cleaned: Leviticus_4.txt
Cleaned: Leviticus_5.txt
Cleaned: Leviticus_6.txt
Cleaned: Leviticus_7.txt
Cleaned: Leviticus_8.txt
Cleaned: Leviticus_9.txt
Cleaned Verses Leviticus_1.txt
Cleaned Verses Leviticus_10.txt
Cleaned Verses Leviticus_11.txt
Cleaned Verses Leviticus_12.txt
Cleaned Verses Leviticus_13.txt
Cleaned Verses Leviticus_14.txt
Cleaned Verses Leviticus_15.txt
Cleaned Verses Leviticus_16.txt
Cleaned Verses Leviticus_17.txt
Cleaned Verses Levit

### Count Cleaned Corpus Size ###
I created a count corpus function that will iterate through every book in every language folder and counts every word by determining any continuous sequence of characters separated by a whitespace. Its a rough estimate for corpus size.

In [33]:
cleaned_folder = "Bible/Cleaned/"
sizes = count_corpus_size(cleaned_folder)

for lang, count in sizes.items():
    print(f"{lang}: {count} words")

Bikol: 88171 words
Cebuano: 103101 words
English: 99454 words
Hiligaynon: 97012 words
Ilokano: 77601 words
Kapampangan: 91813 words
Pangasinense: 87787 words
Spanish: 88456 words
Tagalog: 98573 words
Waray: 97387 words


#### Corpus Size Per Language ###
- Bikol: 88,171 words
- Cebuano: 103,101 words
- English: 99,454 words
- Hiligaynon: 97,012 words
- Ilokano: 77,601 words
- Kapampangan: 91,813 words
- Pangasinense: 87,787 words
- Spanish: 88,456 words
- Tagalog: 98,573 words
- Waray: 97,387 words

Total: 929,355 words

### Segmenting Cleaned Books By Verse ###

In this part, ill be segmenting all the cleaned books I have from the previous code to their respective CSV files as it is part of one of the requirements

In [50]:
#Bikol
segment_by_verse("Bible/Cleaned/Bikol/Exodus")
segment_by_verse("Bible/Cleaned/Bikol/Genesis")
segment_by_verse("Bible/Cleaned/Bikol/Leviticus")
csvs_to_excel(["Bible/Cleaned/Bikol/Exodus","Bible/Cleaned/Bikol/Genesis","Bible/Cleaned/Bikol/Leviticus"],"Bible/Excel/Bikol_Cleaned.xlsx")


Saved 1213 verses to Bible\CSV\Bikol_Exodus.csv
Saved 1533 verses to Bible\CSV\Bikol_Genesis.csv
Saved 859 verses to Bible\CSV\Bikol_Leviticus.csv
Saved Excel workbook to Bible/Excel/Bikol_Cleaned.xlsx


In [4]:
#Cebuano
segment_by_verse("Bible/Cleaned/Cebuano/Exodus")
segment_by_verse("Bible/Cleaned/Cebuano/Genesis")
segment_by_verse("Bible/Cleaned/Cebuano/Leviticus")
csvs_to_excel(["Bible/Cleaned/Cebuano/Exodus","Bible/Cleaned/Cebuano/Genesis","Bible/Cleaned/Cebuano/Leviticus"],"Bible/Excel/Cebuano_Cleaned.xlsx")


Saved 1213 verses to Bible\CSV\Cebuano_Exodus.csv
Saved 1533 verses to Bible\CSV\Cebuano_Genesis.csv
Saved 859 verses to Bible\CSV\Cebuano_Leviticus.csv
Saved Excel workbook to Bible/Excel/Cebuano_Cleaned.xlsx


In [48]:
#English
segment_by_verse("Bible/Cleaned/English/Exodus")
segment_by_verse("Bible/Cleaned/English/Genesis")
segment_by_verse("Bible/Cleaned/English/Leviticus")
csvs_to_excel(["Bible/Cleaned/English/Exodus","Bible/Cleaned/English/Genesis","Bible/Cleaned/English/Leviticus"],"Bible/Excel/English_Cleaned.xlsx")


Saved 1210 verses to Bible\CSV\English_Exodus.csv
Saved 1533 verses to Bible\CSV\English_Genesis.csv
Saved 859 verses to Bible\CSV\English_Leviticus.csv
Saved Excel workbook to Bible/Excel/English_Cleaned.xlsx


In [47]:
#Hiligaynon
segment_by_verse("Bible/Cleaned/Hiligaynon/Exodus")
segment_by_verse("Bible/Cleaned/Hiligaynon/Genesis")
segment_by_verse("Bible/Cleaned/Hiligaynon/Leviticus")
csvs_to_excel(["Bible/Cleaned/Hiligaynon/Exodus","Bible/Cleaned/Hiligaynon/Genesis","Bible/Cleaned/Hiligaynon/Leviticus"],"Bible/Excel/Hiligaynon_Cleaned.xlsx")

Saved 1095 verses to Bible\CSV\Hiligaynon_Exodus.csv
Saved 1234 verses to Bible\CSV\Hiligaynon_Genesis.csv
Saved 752 verses to Bible\CSV\Hiligaynon_Leviticus.csv
Saved Excel workbook to Bible/Excel/Hiligaynon_Cleaned.xlsx


In [46]:
#Ilokano
segment_by_verse("Bible/Cleaned/Ilokano/Exodus")
segment_by_verse("Bible/Cleaned/Ilokano/Genesis")
segment_by_verse("Bible/Cleaned/Ilokano/Leviticus")
csvs_to_excel(["Bible/Cleaned/Ilokano/Exodus","Bible/Cleaned/Ilokano/Genesis","Bible/Cleaned/Ilokano/Leviticus"],"Bible/Excel/Ilokano_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Ilokano_Exodus.csv
Saved 1533 verses to Bible\CSV\Ilokano_Genesis.csv
Saved 859 verses to Bible\CSV\Ilokano_Leviticus.csv
Saved Excel workbook to Bible/Excel/Ilokano_Cleaned.xlsx


In [45]:
#Kapampangan
segment_by_verse("Bible/Cleaned/Kapampangan/Exodus")
segment_by_verse("Bible/Cleaned/Kapampangan/Genesis")
segment_by_verse("Bible/Cleaned/Kapampangan/Leviticus")
csvs_to_excel(["Bible/Cleaned/Kapampangan/Exodus","Bible/Cleaned/Kapampangan/Genesis","Bible/Cleaned/Kapampangan/Leviticus"],"Bible/Excel/Kapampangan_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Kapampangan_Exodus.csv
Saved 1533 verses to Bible\CSV\Kapampangan_Genesis.csv
Saved 859 verses to Bible\CSV\Kapampangan_Leviticus.csv
Saved Excel workbook to Bible/Excel/Kapampangan_Cleaned.xlsx


In [44]:
#Pangasinense
segment_by_verse("Bible/Cleaned/Pangasinense/Exodus")
segment_by_verse("Bible/Cleaned/Pangasinense/Genesis")
segment_by_verse("Bible/Cleaned/Pangasinense/Leviticus")
csvs_to_excel(["Bible/Cleaned/Pangasinense/Exodus","Bible/Cleaned/Pangasinense/Genesis","Bible/Cleaned/Pangasinense/Leviticus"],"Bible/Excel/Pangasinense_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Pangasinense_Exodus.csv
Saved 1533 verses to Bible\CSV\Pangasinense_Genesis.csv
Saved 860 verses to Bible\CSV\Pangasinense_Leviticus.csv
Saved Excel workbook to Bible/Excel/Pangasinense_Cleaned.xlsx


In [43]:
#Spanish
segment_by_verse("Bible/Cleaned/Spanish/Exodus")
segment_by_verse("Bible/Cleaned/Spanish/Genesis")
segment_by_verse("Bible/Cleaned/Spanish/Leviticus")
csvs_to_excel(["Bible/Cleaned/Spanish/Exodus","Bible/Cleaned/Spanish/Genesis","Bible/Cleaned/Spanish/Leviticus"],"Bible/Excel/Spanish_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Spanish_Exodus.csv
Saved 1533 verses to Bible\CSV\Spanish_Genesis.csv
Saved 859 verses to Bible\CSV\Spanish_Leviticus.csv
Saved Excel workbook to Bible/Excel/Spanish_Cleaned.xlsx


In [42]:
#Tagalog
segment_by_verse("Bible/Cleaned/Tagalog/Exodus")
segment_by_verse("Bible/Cleaned/Tagalog/Genesis")
segment_by_verse("Bible/Cleaned/Tagalog/Leviticus")
csvs_to_excel(["Bible/Cleaned/Tagalog/Exodus","Bible/Cleaned/Tagalog/Genesis","Bible/Cleaned/Tagalog/Leviticus"],"Bible/Excel/Tagalog_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Tagalog_Exodus.csv
Saved 1533 verses to Bible\CSV\Tagalog_Genesis.csv
Saved 859 verses to Bible\CSV\Tagalog_Leviticus.csv
Saved Excel workbook to Bible/Excel/Tagalog_Cleaned.xlsx


In [41]:
#Waray
segment_by_verse("Bible/Cleaned/Waray/Exodus")
segment_by_verse("Bible/Cleaned/Waray/Genesis")
segment_by_verse("Bible/Cleaned/Waray/Leviticus")
csvs_to_excel(["Bible/Cleaned/Waray/Exodus","Bible/Cleaned/Waray/Genesis","Bible/Cleaned/Waray/Leviticus"],"Bible/Excel/Waray_Cleaned.xlsx")

Saved 1213 verses to Bible\CSV\Waray_Exodus.csv
Saved 1533 verses to Bible\CSV\Waray_Genesis.csv
Saved 859 verses to Bible\CSV\Waray_Leviticus.csv
Saved Excel workbook to Bible/Excel/Waray_Cleaned.xlsx


### Creating the Parallel Corpora of Languages ###
I created a function in util.py to automatically create an xlsx file after strictly 1:1 matching the languages I pass in, The languages I'll be passing in are: Spanish, Tagalog, Cebuano and Bikolano.

Since the projects require 3 + N, I'll be creating 4 unique combinations since I am working on this solo.

In [19]:
csv_files = {
    "Bikol": ["Bible/CSV/Bikol_Exodus.csv","Bible/CSV/Bikol_Genesis.csv","Bible/CSV/Bikol_Leviticus.csv"],
    "Tagalog": ["Bible/CSV/Tagalog_Exodus.csv","Bible/CSV/Tagalog_Genesis.csv","Bible/CSV/Tagalog_Leviticus.csv"],
    "Cebuano": ["Bible/CSV/Cebuano_Exodus.csv","Bible/CSV/Cebuano_Genesis.csv","Bible/CSV/Cebuano_Leviticus.csv"],
    "Spanish": ["Bible/CSV/Spanish_Exodus.csv","Bible/CSV/Spanish_Genesis.csv","Bible/CSV/Spanish_Leviticus.csv"]
}
pairs = [
    ("Spanish", "Tagalog"),
    ("Bikol", "Cebuano"),
    ("Tagalog", "Cebuano"),
    ("Spanish", "Bikol")
]

create_parallel_corpora(csv_files, pairs, "Bible/Parallel_Corpora")

Saved: Bible/Parallel_Corpora\Spanish_Tagalog_parallel.csv
Saved: Bible/Parallel_Corpora\Bikol_Cebuano_parallel.csv
Saved: Bible/Parallel_Corpora\Tagalog_Cebuano_parallel.csv
Saved: Bible/Parallel_Corpora\Spanish_Bikol_parallel.csv


In [20]:
csv_files = [
    "Bible/Parallel_Corpora/Bikol_Cebuano_parallel.csv",
    "Bible/Parallel_Corpora/Spanish_Bikol_parallel.csv",
    "Bible/Parallel_Corpora/Spanish_Tagalog_parallel.csv",
    "Bible/Parallel_Corpora/Tagalog_Cebuano_parallel.csv"
]

output_excel = "Bible/Parallel_Corpora/Parallel_Corpora.xlsx"

merge_csvs_to_excel(csv_files, output_excel)

Saved Excel workbook to Bible/Parallel_Corpora/Parallel_Corpora.xlsx
